File: auth-service/src/main/java/com/eirsteir/coffeewithme/authservice/web/request/RegistrationRequest.java
package com.eirsteir.coffeewithme.authservice.web.request;

import com.eirsteir.coffeewithme.authservice.validation.ValidPassword;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import lombok.experimental.Accessors;

@Getter
@Setter
@ToString
@Accessors(chain = true)
@NoArgsConstructor
public class RegistrationRequest {

  @Email(message = "Must be a well formed email")
  @NotBlank(message = "Email is required")
  private String email;

  @ValidPassword private String password;

  @NotBlank(message = "Name is required")
  private String name;
}

File: auth-service/src/main/kotlin/com/eirsteir/coffeewithme/authservice/web/request/RegistrationRequest.kt
package com.eirsteir.coffeewithme.authservice.web.request

import com.eirsteir.coffeewithme.authservice.validation.ValidPassword
import javax.validation.constraints.Email
import javax.validation.constraints.NotBlank


data class RegistrationRequest(
    @get:Email(message = "Must be a well formed email")
    @get:NotBlank(message = "Email is required")
    val email: String,
    @field:ValidPassword
    val password: String,
    @get:NotBlank(message = "Name is required")
    val name:  String
)